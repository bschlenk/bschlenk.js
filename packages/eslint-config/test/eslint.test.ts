import { ESLint } from 'eslint'
import path from 'path'
import { describe, expect, test } from 'vitest'

describe('eslint', () => {
  const eslint = new ESLint({
    cwd: import.meta.dirname,
    ignore: false,
  })

  test('...', async () => {
    const results = await eslint.lintFiles(
      path.join(import.meta.dirname, './bad.tsx')
    )
    for (const r of results) {
      delete r.source

      // remove suppressed messages
      r.messages = r.messages.filter((m) => !('suppressions' in m))
      // @ts-expect-error safe to delete
      delete r.suppressedMessages

      for (const m of r.messages) {
        if ('fix' in m && m.fix && 'text' in m.fix) {
          // @ts-expect-error safe to delete
          delete m.fix.text
        }
      }
      r.filePath = path.relative(__dirname, r.filePath)
    }
    expect(results).toMatchSnapshot()
  })
})
