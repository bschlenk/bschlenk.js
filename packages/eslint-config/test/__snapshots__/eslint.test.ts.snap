// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`eslint > ... 1`] = `
[
  {
    "errorCount": 18,
    "fatalErrorCount": 0,
    "filePath": "bad.tsx",
    "fixableErrorCount": 11,
    "fixableWarningCount": 0,
    "messages": [
      {
        "column": 1,
        "endColumn": 20,
        "endLine": 5,
        "fix": {
          "range": [
            56,
            123,
          ],
        },
        "line": 3,
        "message": "Run autofix to sort these imports!",
        "messageId": "sort",
        "nodeType": null,
        "ruleId": "simple-import-sort/imports",
        "severity": 2,
      },
      {
        "column": 18,
        "endColumn": 24,
        "endLine": 3,
        "fix": {
          "range": [
            80,
            104,
          ],
        },
        "line": 3,
        "message": "'path' imported multiple times.",
        "nodeType": "Literal",
        "ruleId": "import/no-duplicates",
        "severity": 2,
      },
      {
        "column": 18,
        "endColumn": 24,
        "endLine": 4,
        "line": 4,
        "message": "'path' imported multiple times.",
        "nodeType": "Literal",
        "ruleId": "import/no-duplicates",
        "severity": 2,
      },
      {
        "column": 1,
        "fix": {
          "range": [
            123,
            123,
          ],
        },
        "line": 5,
        "message": "Expected 1 empty line after import statement not followed by another import.",
        "nodeType": null,
        "ruleId": "import/newline-after-import",
        "severity": 2,
      },
      {
        "column": 5,
        "endColumn": 6,
        "endLine": 11,
        "fix": {
          "range": [
            201,
            213,
          ],
        },
        "line": 11,
        "message": "'x' is never reassigned. Use 'const' instead.",
        "messageId": "useConst",
        "nodeType": "Identifier",
        "ruleId": "prefer-const",
        "severity": 2,
      },
      {
        "column": 1,
        "fix": {
          "range": [
            243,
            277,
          ],
        },
        "line": 14,
        "message": "Unused eslint-disable directive (no problems were reported from 'eqeqeq').",
        "nodeType": null,
        "ruleId": null,
        "severity": 2,
      },
      {
        "column": 18,
        "endColumn": 47,
        "endLine": 20,
        "fix": {
          "range": [
            359,
            388,
          ],
        },
        "line": 20,
        "message": "Curly braces are unnecessary here.",
        "messageId": "unnecessaryCurly",
        "nodeType": "JSXExpressionContainer",
        "ruleId": "react/jsx-curly-brace-presence",
        "severity": 2,
      },
      {
        "column": 48,
        "endColumn": 76,
        "endLine": 20,
        "fix": {
          "range": [
            389,
            417,
          ],
        },
        "line": 20,
        "message": "Curly braces are unnecessary here.",
        "messageId": "unnecessaryCurly",
        "nodeType": "JSXExpressionContainer",
        "ruleId": "react/jsx-curly-brace-presence",
        "severity": 2,
      },
      {
        "column": 6,
        "endColumn": 12,
        "endLine": 28,
        "fix": {
          "range": [
            558,
            597,
          ],
        },
        "line": 28,
        "message": "Use an \`interface\` instead of a \`type\`.",
        "messageId": "interfaceOverType",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/consistent-type-definitions",
        "severity": 2,
      },
      {
        "column": 62,
        "endColumn": 64,
        "endLine": 37,
        "line": 37,
        "message": "Unexpected empty function 'inlineObjectType'.",
        "messageId": "unexpected",
        "nodeType": "FunctionDeclaration",
        "ruleId": "@typescript-eslint/no-empty-function",
        "severity": 2,
      },
      {
        "column": 18,
        "endColumn": 27,
        "endLine": 42,
        "fix": {
          "range": [
            838,
            849,
          ],
        },
        "line": 42,
        "message": "Empty components are self-closing",
        "messageId": "notSelfClosing",
        "nodeType": "JSXOpeningElement",
        "ruleId": "react/self-closing-comp",
        "severity": 2,
      },
      {
        "column": 10,
        "endColumn": 11,
        "endLine": 54,
        "line": 54,
        "message": "'_' is defined but never used.",
        "messageId": "unusedVar",
        "nodeType": null,
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 2,
      },
      {
        "column": 5,
        "endColumn": 6,
        "endLine": 59,
        "fix": {
          "range": [
            1122,
            1136,
          ],
        },
        "line": 59,
        "message": "'y' is never reassigned. Use 'const' instead.",
        "messageId": "useConst",
        "nodeType": "Identifier",
        "ruleId": "prefer-const",
        "severity": 2,
      },
      {
        "column": 5,
        "endColumn": 6,
        "endLine": 59,
        "line": 59,
        "message": "'y' is assigned a value but never used. Allowed unused vars must match /^_/u.",
        "messageId": "unusedVar",
        "nodeType": null,
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 2,
      },
      {
        "column": 3,
        "endColumn": 12,
        "endLine": 62,
        "line": 62,
        "message": "'unusedArg' is defined but never used. Allowed unused args must match /^_/u.",
        "messageId": "unusedVar",
        "nodeType": null,
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 2,
      },
      {
        "column": 3,
        "endColumn": 16,
        "endLine": 64,
        "line": 64,
        "message": "'anotherUnused' is defined but never used. Allowed unused args must match /^_/u.",
        "messageId": "unusedVar",
        "nodeType": null,
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 2,
      },
      {
        "column": 1,
        "endColumn": 7,
        "endLine": 86,
        "line": 86,
        "message": "Unexpected use of 'Buffer'. Use Uint8Array instead.",
        "messageId": "customMessage",
        "nodeType": "Identifier",
        "ruleId": "no-restricted-globals",
        "severity": 2,
      },
      {
        "column": 1,
        "endColumn": 32,
        "endLine": 89,
        "fix": {
          "range": [
            1735,
            1766,
          ],
        },
        "line": 89,
        "message": "Run autofix to sort these exports!",
        "messageId": "sort",
        "nodeType": null,
        "ruleId": "simple-import-sort/exports",
        "severity": 2,
      },
    ],
    "usedDeprecatedRules": [],
    "warningCount": 0,
  },
]
`;
